src/transformers/generation_utils.py

```diff
@@ -370,10 +370,16 @@ def adjust_logits_during_generation(self, logits: torch.FloatTensor, **kwargs) -
         """
         return logits
 
-    def _prepare_input_ids_for_generation(self, bos_token_id: int) -> torch.LongTensor:
-        if bos_token_id is None:
-            raise ValueError("`bos_token_id` has to be defined when no `input_ids` are provided.")
-        return torch.ones((1, 1), dtype=torch.long, device=self.device) * bos_token_id
+    def _prepare_input_ids_for_generation(
+        self, bos_token_id: Optional[int], encoder_outputs: Optional[ModelOutput]
+    ) -> torch.LongTensor:
+        if encoder_outputs is None:
+            if bos_token_id is None:
+                raise ValueError("`bos_token_id` has to be defined when no `input_ids` are provided.")
+            return torch.ones((1, 1), dtype=torch.long, device=self.device) * bos_token_id
+        else:
+            shape = encoder_outputs.last_hidden_state.size()[:-1]
+            return torch.ones(shape, dtype=torch.long, device=self.device) * -100
```

Comment:
The `* -100` merits a comment here I think -> it's not directly obvious why we multiply by -100 IMO

Reply:
I would also be in favor of having comments across this `_prepare_input_ids_for_generation` method as it is quite cryptic in its proposed state.

Reply:
Thanks for the suggestion. I've added a comment now. I admit it's a little bit hacky because the encoder won't be run and hence the `input_ids` won't be used; the `input_ids` is created just to satisfy the subsequent processing, which depends on it for the input shape. The value -100 is chosen so that if the `input_ids` were used for actual encoding, an error would be thrown.

I notice there is a relatively new argument `encoder_no_repeat_ngram_size`, which would make the logits processor dependent on the encoder input_ids. Because the dummy input_ids would be all -100, which is impossible for `decoder_input_ids`, this won't have any unexpected effect.